fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/source/reader/FlinkSourceSplitReader.java

```diff
@@ -145,6 +149,12 @@ public RecordsWithSplitIds<RecordAndPos> fetch() throws IOException {
                                 new HashSet<>(emptyLogSplits), flinkSourceReaderMetrics);
                 emptyLogSplits.clear();
                 return records;
+            } else if (!removedSplits.isEmpty()) {
```

Comment:
I think we can check `!removedSplits.isEmpty()` at the beginning so that the flink enumerate can remove the state of removed splits as soon as possible.

Reply:
Yes, we can move the code block at the beginning of `fetch()` method, because the `currentBoundedSplitReader` may still work in progress.

Reply:
That makes sense. I have moved it to the beginning of `fetch`